src/webpack-plugin.ts

```diff
@@ -31,10 +48,33 @@ class VueAutoRoutingPlugin {
 
       fs.writeFileSync(to, code)
     }
+    const generateIndex = (options: Options[]) => {
+      let importCode = '', exportCode = '';
+      options.forEach(option => {
+        importCode += `import ${option.pageName} from './${option.pageName}';\n`
+        exportCode += `${option.pageName},`
+      })
+      return `${importCode}export{${exportCode}}`
+    }
 
     compiler.hooks.thisCompilation.tap(pluginName, compilation => {
       try {
-        generate()
+        if (Object.prototype.toString.call(this.options) === '[object Object]') {
+          generate(this.options as Options);
+        } else if (Array.isArray(this.options)) {
+          this.options.forEach(option => {
+            generate(option)
+          })
+          const indexPath = path.resolve(__dirname, '../index.js');
+          const code = generateIndex(this.options)
+          if (fs.existsSync(indexPath) &&
+            fs.readFileSync(indexPath, 'utf8').trim() === code.trim()) {
```

Comment:
We have to check every generated file including route files for each page.

Reply:
fixed